bin/lcov

```diff
@@ -477,14 +479,15 @@ elsif (@opt_summary)
 
 temp_cleanup();
 
+our $exit_code = 0;
 if (defined($ln_overall_found)) {
-	print_overall_rate(1, $ln_overall_found, $ln_overall_hit,
-			   1, $fn_overall_found, $fn_overall_hit,
-			   1, $br_overall_found, $br_overall_hit);
+	$exit_code = print_overall_rate(1, $ln_overall_found, $ln_overall_hit,
+					1, $fn_overall_found, $fn_overall_hit,
+					1, $br_overall_found, $br_overall_hit);
```

Comment:
This logic should not be added to `print_overall_rate` - the function name does not imply any kind of checking. Also adding it here enables the check for more than just `--summary` which is not what the help text/man page implies.

Instead add a new function named something like `check_rates()` which returns the exit code based on the actual coverage rates. This function should then only be called from withing the `if (@opt_summary)` branch.

Reply:
Done.